sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -848,4 +852,40 @@ object DateTimeUtils {
     val sinceEpoch = BigDecimal(timestamp) / MICROS_PER_SECOND + offset
     new Decimal().set(sinceEpoch, 20, 6)
   }
+
+  def currentDate(zoneId: ZoneId): SQLDate = localDateToDays(LocalDate.now(zoneId))
+
+  private val specialDate = """(EPOCH|NOW|TODAY|TOMORROW|YESTERDAY)\p{Blank}*(.*)""".r
+  /**
+   * Converts notational shorthands that are converted to ordinary dates.
+   * @param input - left trimmed string
+   * @param zoneId - zone identifier used to get the current date.
+   * @return some of days since the epoch if the conversion completed successfully otherwise None.
+   */
+  def convertSpecialDate(input: String, zoneId: ZoneId): Option[SQLDate] = {
+    def isValidZoneId(z: String): Boolean = {
+      z == "" || Try { getZoneId(z) }.isSuccess
```

Comment:
Better to avoid the `Try` use?

Reply:
Why?

Reply:
Just for the codestyle? https://github.com/databricks/scala-style-guide#exception

Reply:
I don't think the reasons from the style guide are applicable here. And code looks slightly ugly. Maybe you have better implementation than:
```scala
    def isValidZoneId(z: String): Boolean = {
      if (z == "") return true
      try {
        getZoneId(z)
        true
      } catch {
        case NonFatal(_) => false
      }
    }
```